src/python/tests/test_initialisation.py

```diff
@@ -0,0 +1,187 @@
+                    "collisional_ionisation_rates", "recombination_cooling_rates", "bremsstrahlung_cooling_rates",\
+                    "h2_h_cooling_rate", "photoelectric_heating"]
+    for parameter in parameters:
+        print(parameter + ":", getattr(my_chemistry, parameter))
+
+#* Function which tells you unique order of magnitude discrepancies for a given rate between two hdf5 files.
+def oom_discrepancies(rateName, parameterSet, rateFile1, rateFile2):
+    name = rateName + f'_{parameterSet}'
+    rates1 = rateFile1[name]
+    rates2 = rateFile2[name]
+
+    uniqueDiscrepancies = {}
+    #Check each rate coefficient and calculate OOM discrepancy.
+    for i in range(len(rates1)):
+        if rates1[i] == 0 or rates2[i] == 0:
+            None
+        else:
+            relDisc = abs(rates1[i] - rates2[i]) / rates1[i]
+            if relDisc == 0:
+                None
+            else:
+                DiscOOM = np.floor(np.log10(relDisc))
+                if f'e^{DiscOOM}' not in uniqueDiscrepancies:
+                    uniqueDiscrepancies[f'e^{DiscOOM}'] = []
+                uniqueDiscrepancies[f'e^{DiscOOM}'].append(i)
+
+    return uniqueDiscrepancies
+
+
+#* Function which tests that the rates have been initialised correctly for each parameter set.
+def test_rate_initialisation(printParameters=False, printOOMdiscrepanices=False):
+    """
+    Test that the rate tables are initialized correctly.
+
+    If writeHDF5 is set to true, all rate coefficients calculated in the initialisation routine will 
+    be saved into a hdf5 file. This should be used for testing and is not meant for frequent use.
+
+    printParameters (bool) --> If set to True will print the parameter settings for each parameter set.
+    """
+
+    #* Navigate to the directory where the file is located.
+    filePath = os.path.abspath(__file__)
+    dirPath = os.path.dirname(filePath)
+    os.chdir(dirPath)
+
+    #* List of all rate variable names which will be checked.
+    testRates = ["k1", "k2", "k3", "k4", "k5", "k6", "k7", "k8", "k9", "k10", "k11", "k12", "k13",
+                 "k14", "k15", "k16","k17", "k18", "k19", "k20", "k21", "k22", "k23", "k24", "k25",
+                 "k26", "k27", "k28", "k29", "k30", "k31", "k50", "k51", "k52", "k53", "k54", "k55",
+                 "k56", "k57", "k58", "n_cr_n", "n_cr_d1", "n_cr_d2", "ceHI", "ceHeI", "ceHeII",
+                 "ciHI", "ciHeI", "ciHeIS", "ciHeII", "reHII", "reHeII1", "reHeII2","reHeIII", "brem",
+                 "hyd01k", "h2k01", "vibh", "roth", "rotl", "HDlte", "HDlow","cieco", "GAHI", "GAH2",
+                 "GAHe", "GAHp", "GAel", "H2LTE", "k13dd", "h2dust"]
+
+    #* Calculate rates for each parameter set and write to hdf5 file
+    #If file already exists delete it so that the new one can be created.
+    if os.path.exists("initialised_rates.h5"):
+        os.remove("initialised_rates.h5")
+    #Create and open file.
+    f = h5py.File("initialised_rates.h5", "w-")
```

Comment:
Change the access mode in `h5py.File` from `w-` to `w` and you won't have to manually remove the file if it already exists.